zanata-war/src/main/java/org/zanata/dao/DatabaseSpecific.java

```diff
@@ -0,0 +1,18 @@
+package org.zanata.dao;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marker annotation to mark methods that will only work in certain database.
+ */
+@Target({ElementType.METHOD, ElementType.LOCAL_VARIABLE, ElementType.TYPE, ElementType.PARAMETER})
+@Retention(RetentionPolicy.SOURCE)
+public @interface DatabaseSpecific {
```

Comment:
That's what I mean, maybe we can just add an extra param to that annotation. Something like:
```java
@NativeQuery( description="This is a native", specificTo="MySQL" )
```
It's just a marker annotation with some info, so no need to create more classes for it IMO.